### ISSUE TITLE
Replace setuptools with scikit-build

### ISSUE THREAD

**Issue by maxbachmann** [maxbachmann/rapidfuzz#155]
It would be interesting to replace setuptools, since the SIMD support from https://github.com/maxbachmann/RapidFuzz/issues/127 would require us to compile parts of the projects with different compiler flags. This would be a lot simpler to implement in CMake. I tried to switch to scikit-build in https://github.com/maxbachmann/RapidFuzz/pull/153. However I did run into issues with PyPy and Python2.7, which I could not resolve. From https://github.com/scikit-build/scikit-build/issues/506 it appears, that PyPy is not really supported well at this point.

**Comment by henryiii** [comment]
FYI, https://iscinumpy.gitlab.io/post/scikit-build-proposal - it's planned (but might depend on funding, which might depend on how many collaborators and use cases are available, etc).

**Comment by maxbachmann** [comment]
For me scikit-build is definetly a big improvement, since setuptools becomes a pain for non trivial build. I will drop Python2.7 in v2.0.0 and might release it without PyPy wheels if I can not get it to work, since I do not think many users use this library from PyPy. However it would be nice to get PyPy working with scikit-build at some point.

If you need any help testing just let me know.

**Comment by maxbachmann** [comment]
RapidFuzz is now using scikit-build. For now this requires a workaround for Cython and drops PyPy support. However things should improve when https://github.com/scikit-build/scikit-build/discussions/601 gets implemented.

**Comment by henryiii** [comment]
Feel free to let us know if you have problems!

**Comment by henryiii** [comment]
FYI, `with io.open('README.md', 'rt', encoding="utf8") as f:` in setup.py -  you don't need `io.open` anymore with Python 3, regular `open` is fine. pyupgrade (ideally with pre-commit) will cleanup these sorts of things for you. See https://scikit-hep.org/developer/style

**Comment by maxbachmann** [comment]
> you don't need io.open anymore with Python 3

Thanks for mentioning. This is the first release without Python2.7 support. In fact other than type hints the setup.py is the only pure Python file of the project. I did just try `pyupgrade`, which successfully transforms `io.open` to `open`. However it does not remove the then unused `import io` statement ...
I am looking forward to writing all of this in the `pyproject.toml` file and getting rid of `setup.py` ;)

> Feel free to let us know if you have problems!

My problems so far are:
1) I have problems compiling the project for PyPy on some platforms with scikit-build. I think this was mostly problematic for Windows: https://github.com/maxbachmann/RapidFuzz/runs/4666969968?check_suite_focus=true

Which fails with
```
     Call Stack (most recent call first):
      C:/Users/runneradmin/AppData/Local/Temp/pip-build-env-j5_mpiwu/overlay/site-packages/cmake/data/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
      C:/Users/runneradmin/AppData/Local/Temp/pip-build-env-j5_mpiwu/overlay/site-packages/cmake/data/share/cmake-3.22/Modules/FindPythonLibs.cmake:310 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
      C:/Users/runneradmin/AppData/Local/Temp/pip-build-env-j5_mpiwu/overlay/site-packages/skbuild/resources/cmake/FindNumPy.cmake:44 (find_package)
      CMakeLists.txt:12 (find_package)
```
For now I dropped support for PyPy, since it is not used much anyways

2) I did not manage to add the Cython targets to the top level `CMakeLists.txt`: https://github.com/maxbachmann/RapidFuzz/blob/main/CMakeLists.txt and had to create one in the subdirectory, so the cython file is found: https://github.com/maxbachmann/RapidFuzz/blob/main/src/CMakeLists.txt. This appears to be a known issue: https://github.com/scikit-build/scikit-build/issues/543 and my current solution works for me (I would prefer being able to write them in the top level `CMakeLists.txt` in the long run)


Other than this the combination of scikit-build/cibuildwheels/github actions is awesome. They make building binary packages a lot less trouble.
